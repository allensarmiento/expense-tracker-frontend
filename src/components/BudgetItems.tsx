import styled from 'styled-components';
import BudgetItem, { Item } from './BudgetItem';

const Items = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  position: relative;
  height: 100%;
  border: 1px solid var(--color-grey-dark-1);
  border-radius: .4rem;
  font-size: 2rem;
  overflow: auto;
`;

const Header = styled.header`
  display: grid;
  grid-template-columns: minmax(max-content, 1fr)
                         minmax(20rem, 2fr)
                         minmax(40rem, 2fr)
                         minmax(10rem, 1fr)
                         minmax(4rem,  min-content);
  grid-column-gap: 2rem;
  position: sticky;
  top: 0;
  left: 0;
  padding: 1rem 2rem;
  background: #fff;
  font-weight: 600;
`;

const Heading = styled.span``;

type BudgetItemsProps = {
  items: Array<Item>;
};

const BudgetItems = ({ items }: BudgetItemsProps) => {
  return (
    <Items>
      <Header>
        <Heading>Date</Heading>
        <Heading>Category</Heading>
        <Heading>Description</Heading>
        <Heading>Amount</Heading>
      </Header>
      {items.map((item, index) => (
        <BudgetItem
          key={index}
          item={item}
        />
      ))}
    </Items>
  );
};

export default BudgetItems;
