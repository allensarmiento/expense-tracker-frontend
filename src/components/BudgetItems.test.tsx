import { shallow } from 'enzyme';
import BudgetItems from './BudgetItems';

it('renders without crashing', () => {
  expect(shallow(
    <BudgetItems items={[{
      date: '2021-02-22',
      type: 'expense',
      category: 'food',
      description: 'Pizza',
      amount: 19.99,
      fixed: 'yes',
      paid: 'yes'
    }]} />
  )).toMatchSnapshot();
});
