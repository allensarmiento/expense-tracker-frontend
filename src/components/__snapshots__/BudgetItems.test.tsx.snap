// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<styled.div>
  <styled.header>
    <styled.span>
      Date
    </styled.span>
    <styled.span>
      Category
    </styled.span>
    <styled.span>
      Description
    </styled.span>
    <styled.span>
      Amount
    </styled.span>
  </styled.header>
  <BudgetItem
    item={
      Object {
        "amount": 19.99,
        "category": "food",
        "date": "2021-02-22",
        "description": "Pizza",
        "fixed": "yes",
        "paid": "yes",
        "type": "expense",
      }
    }
    key="0"
  />
</styled.div>
`;
